android/src/main/java/com/swmansion/reanimated/keyboard/KeyboardAnimationCallback.java

```diff
@@ -31,14 +34,31 @@ public WindowInsetsAnimationCompat.BoundsCompat onStart(
   public WindowInsetsCompat onProgress(
       @NonNull WindowInsetsCompat insets,
       @NonNull List<WindowInsetsAnimationCompat> runningAnimations) {
+    boolean isAnyKeyboardAnimation = false;
+    for (WindowInsetsAnimationCompat animation : runningAnimations) {
+      if (isKeyboardAnimation(animation)) {
+        isAnyKeyboardAnimation = true;
+        break;
+      }
+    }
+    if (!isAnyKeyboardAnimation) {
+      return insets;
+    }
     keyboard.updateHeight(insets);
     notifyAboutKeyboardChange.call();
     return insets;
   }
 
   @Override
   public void onEnd(@NonNull WindowInsetsAnimationCompat animation) {
+    if (!isKeyboardAnimation(animation)) {
+      return;
+    }
     keyboard.onAnimationEnd();
     notifyAboutKeyboardChange.call();
   }
+
+  private boolean isKeyboardAnimation(@NonNull WindowInsetsAnimationCompat animation) {
```

Comment:
This method could be static
```suggestion
  private static boolean isKeyboardAnimation(@NonNull WindowInsetsAnimationCompat animation) {
```